src/meter/Meter.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import FlexView from '../flex/FlexView';
+import find from 'lodash/collection/find';
+import { warn } from '../utils/log';
+
+const Range = t.refinement(t.struct({
+  startValue: t.maybe(t.Number),
+  endValue: t.Number,
+  fillingColor: t.String,
+  labelColor: t.String
+}), r => r.startValue < r.endValue);
+
+const labelFormatter = (currentValue, startValue, endValue) =>(
+  Math.abs((currentValue - startValue) * 100 / (endValue - startValue))+'%'
+);
+/**
+ * ### Renders a Progress Bar
+ */
+@skinnable()
+@props({
+  /**
+   * This is the value provided as input.
+   */
+  currentValue: t.Number,
+  /**
+   * Minimum value.
+   */
+  minValue: t.maybe(t.Number),
+  /**
+   * Maximum value.
+   */
+  maxValue: t.maybe(t.Number),
+  /**
+   * Function as input in which you can define a custom label format.
+   */
+  labelFormatter: t.maybe(t.Function),
+  /**
+   * Array of Object in which you can define startValue, endValue, labelColor, fillingColor.
+   */
+  steps: t.maybe(t.list(Range)),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+})
+export default class Meter extends React.Component{
+
+  static defaultProps = {
+    minValue: 0,
+    maxValue: 100,
+    labelFormatter
+  };
+
+  validateSteps = (steps) => {
+    if(!steps){
+      return false;
+    } else {
+      return steps.reduce(( memo, curr, index, arr ) => {
+        if(index === 1){
+          memo.startValue = this.props.minValue;
+        }
+        curr.startValue = arr[index-1].endValue + 1; //scale to be defined
+        return memo;
+      });
+    }
+  }
```

Comment:
about this validate steps function, just one thing to say for now: IMO it should be part of props validations, i.e. a `refinement` for the `steps` prop type